Product/Production/Services/PatientDiscoveryCore/src/test/java/gov/hhs/fha/nhinc/patientdiscovery/parser/TestPatientDiscoveryMessageHelper.java

```diff
@@ -0,0 +1,345 @@
+        return response;
+    }
+
+    private static PRPAIN201305UV02QUQIMT021001UV01ControlActProcess createControlActProcess() {
+        II subjectId = new II();
+        subjectId.setRoot(ROOT);
+        subjectId.setExtension(EXTENSION);
+
+        CD code = new CD();
+        code.setCode("PRPA_TE201306UV02");
+        code.setCodeSystem("2.16.840.1.113883.1.6");
+
+        PRPAIN201305UV02QUQIMT021001UV01ControlActProcess controlActProcess
+            = new PRPAIN201305UV02QUQIMT021001UV01ControlActProcess();
+        controlActProcess.setClassCode(ActClassControlAct.CACT);
+        controlActProcess.setMoodCode(XActMoodIntentEvent.EVN);
+        controlActProcess.setCode(code);
+        controlActProcess.setQueryByParameter(createQueryParams("Allen", "Wanderson", "M", "09-8-1971", subjectId));
+        return controlActProcess;
+    }
+
+    private static JAXBElement<PRPAMT201306UV02QueryByParameter> createQueryParams(String firstName, String lastName,
+        String gender, String birthTime, II subjectId) {
+
+        II id = new II();
+        id.setRoot(QUERY_ID_ROOT);
+        id.setExtension(QUERY_ID_EXTENSION);
+        CS statusCode = new CS();
+        statusCode.setCode("new");
+
+        PRPAMT201306UV02QueryByParameter params = new PRPAMT201306UV02QueryByParameter();
+        params.setQueryId(id);
+        params.setStatusCode(statusCode);
+        params.setParameterList(createParamList(firstName, lastName, gender, birthTime, subjectId));
+
+        return new JAXBElement<>(new QName("urn:hl7-org:v3", "queryByParameter"),
+            PRPAMT201306UV02QueryByParameter.class, params);
+    }
+
+    private static PRPAMT201306UV02ParameterList createParamList(String firstName, String lastName, String gender,
+        String birthTime, II subjectId) {
+
+        PRPAMT201306UV02ParameterList paramList = new PRPAMT201306UV02ParameterList();
+
+        // Set the Subject Gender Code
+        paramList.getLivingSubjectAdministrativeGender().add(createGender(gender));
+
+        // Set the Subject Birth Time
+        paramList.getLivingSubjectBirthTime().add(createBirthTime(birthTime));
+
+        // Set the Subject Name
+        paramList.getLivingSubjectName().add(createName(firstName, lastName));
+
+        // Set the subject Id
+        paramList.getLivingSubjectId().add(createSubjectId(subjectId));
+
+        return paramList;
+    }
+
+    private static PRPAMT201306UV02LivingSubjectId createSubjectId(II subjectId) {
+        PRPAMT201306UV02LivingSubjectId id = new PRPAMT201306UV02LivingSubjectId();
+        if (subjectId != null) {
+            id.getValue().add(subjectId);
+        }
+
+        return id;
+    }
+
+    private static PRPAMT201306UV02LivingSubjectName createName(String firstName, String lastName) {
+        org.hl7.v3.ObjectFactory factory = new org.hl7.v3.ObjectFactory();
+        ENExplicit name = factory.createENExplicit();
+        List namelist = name.getContent();
```

Comment:
The reason I haven't commented in about 15 minutes is, I spent the last 15 minutes trying to figure out what this function is doing :)  I think this would be a hundred times more readable if you removed "namelist" as a separate variable, and explicitly called "name.getContent()" where needed.

Also, consider removing "factory" variable, and calling in-line (ENExplicity name = new org.hl7......).